core/src/main/java/io/questdb/griffin/engine/functions/lt/LtVarcharStrFunctionFactory.java

```diff
@@ -0,0 +1,120 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.lt;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.PlanSink;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.NegatableBooleanFunction;
+import io.questdb.griffin.engine.functions.UnaryFunction;
+import io.questdb.griffin.engine.functions.constants.BooleanConstant;
+import io.questdb.std.IntList;
+import io.questdb.std.ObjList;
+import io.questdb.std.str.Utf8Sequence;
+import io.questdb.std.str.Utf8String;
+import io.questdb.std.str.Utf8s;
+
+public class LtVarcharStrFunctionFactory implements FunctionFactory {
+
+    @Override
+    public String getSignature() {
+        return "<(SØ)";
+    }
+
+    @Override
+    public boolean isBoolean() {
+        return true;
+    }
+
+    @Override
+    public Function newInstance(
+            int position,
+            ObjList<Function> args,
+            IntList argPositions,
+            CairoConfiguration configuration,
+            SqlExecutionContext sqlExecutionContext
+    ) {
+        final Function a = args.getQuick(0);
+        final Function b = args.getQuick(1);
+        if (a.isConstant() && !b.isConstant()) {
+            CharSequence constValue = a.getStrA(null);
+            if (constValue == null) {
+                return BooleanConstant.FALSE;
+            }
+            return new ConstOnLeftFunc(new Utf8String(constValue), b);
+        }
+        if (!a.isConstant() && b.isConstant()) {
+            CharSequence constValue = b.getStrA(null);
+            if (constValue == null) {
+                return BooleanConstant.FALSE;
+            }
+            return new LtStrFunctionFactory.ConstOnRightFunc(a, constValue);
+        }
+        return new LtStrFunctionFactory.Func(a, b);
```

Comment:
Again, if the left had is a runtime constant, we'll end up reencoding the varchar (right hand) for no good reason.

Reply:
This is rather an unpopular function, we didn't have optimized version to handle bind variable as a special case for string type hence I don't see it as a must have for varchar.

Reply:
Well, this at least deserves a comment, so that we don't forget to improve it in future.

Reply:
Commented in [eb12311](https://github.com/questdb/questdb/pull/4308/commits/eb12311eb4702bdfd6783c65317580ca5e4299dc)